components/connectionmanager/src/lib.rs

```diff
@@ -243,8 +246,30 @@ impl ConnectionManager {
             return;
         }
 
-        let mut futures = Vec::with_capacity(active_inbound_len - self.inbound_limit);
-        for peer in active_inbound.choose_multiple(&mut thread_rng(), active_inbound_len - self.inbound_limit) {
+        let peer_overflow_amount = active_inbound_len - self.inbound_limit;
+
+        let mut futures = Vec::with_capacity(peer_overflow_amount);
+
+        // the following peer selection strategy is designed to ensure that an eclipse attacker most outperform
```

Comment:
Maybe typo "most" to "must"?

Reply:
one of my more typical typos :sweat_smile: ..